### PR TITLE
[perf] DeepSeekV3: drop redundant FP32 upcasts in trtllm MoE paths

### PR DESCRIPTION
**Pull request by maocheng23**
## Summary

Drop the explicit `.to(torch.float32)` upcasts on `router_logits` before each `trtllm_*_moe` call for DeepSeekV3 routing. flashinfer ≥ 0.6.8 (`trtllm_fp4_block_scale_moe`, `trtllm_fp8_block_scale_moe`, `trtllm_fp8_per_tensor_scale_moe`) accept BF16 `router_logits` directly, so the explicit FP32 upcasts are no longer required.

flashinfer is pinned at `0.6.11.post1` in `python/pyproject.toml`, well past the 0.6.8 cutoff.

## Changes

**`python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py` — three sites**

1. FP4 block-scale path (`fused_experts_none_to_flashinfer_trtllm_fp4`): drop the `if routing_method_type == DeepSeekV3: router_logits = router_logits.to(torch.float32)` block.
2. FP8 block-scale path (`trtllm_fp8_block_scale_moe_wrapper` call): drop the inline ternary `.to(torch.float32) if DSV3 else router_logits`.
3. FP8 per-tensor path (`trtllm_fp8_per_tensor_scale_moe_wrapper` call): simplify the FP32/BF16 conditional to an unconditional `.to(torch.bfloat16)`.

**`python/sglang/srt/models/deepseek_v2.py`**

`MoEGate.forward` continues to emit FP32 router_logits. A `TODO: will check the dtype to be bf16` is added to the SM100/103 + 256-expert fast path as a marker for the follow-up that would migrate the gate output to BF16 end-to-end (which is the load-bearing change for the actual perf win).

## Why split the BF16 gate change out

`MoEGate`'s output dtype is shared by **all** MoE runner backends, not just `flashinfer_trtllm`. Flipping the gate to BF16 has different implications per backend:

- STANDARD-path topk (`triton`, `deep_gemm`, `flashinfer_cutlass`, `flashinfer_cutedsl`, `cutlass`, `aiter`) internally promotes via `gating_output.float()` (see `topk.py:459, 478`); a BF16 gate would convert that no-op into a real BF16→FP32 cast.
- `triton_kernels`, `marlin`, `flashinfer_mxfp4` (non-fp4-experts) pass `router_logits` directly to their own kernels; per-backend verification is needed.

This PR ships only the trtllm-side cleanup, which is safe with gate=FP32 (the FP4 / FP8 block-scale drops are pure no-ops; the FP8 per-tensor unconditional `to(bfloat16)` is the one site that changes behavior for that specific path). The gate dtype migration is deferred to a follow-up tracked by the TODO.

## Test plan

- [ ] Existing DeepSeek-V3 / V3.2 inference tests pass on SM100+ (B200) and SM90 (H100/H200).
- [ ] Existing FP4 / FP8 MoE accuracy / unit tests pass.

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #26310793188](https://github.com/sgl-project/sglang/actions/runs/26310793188)
Latest PR Test (Extra): :x: [Run #26310793054](https://github.com/sgl-project/sglang/actions/runs/26310793054)

### CODE REVIEW

File: python/sglang/srt/models/deepseek_v2.py [hunk 1]
```diff
@@ -401,17 +401,16 @@ def forward(
                 and _device_sm >= 90
             ):
                 if _device_sm in [100, 103] and self.weight.shape[0] == 256:
-                    # router gemm output float32
                     logits = torch.empty(
                         hidden_states.shape[0],
                         self.weight.shape[0],
                         device=hidden_states.device,
-                        dtype=torch.float32,
+                        dtype=torch.bfloat16,
```
Comment (ch-wan) [inline python/sglang/srt/models/deepseek_v2.py]: It's risky. Add a TODO item here for future discussion.

**Review by ch-wan — APPROVED** [review]

**Review by Qiaolin-Yu — APPROVED** [review]

**Review by b8zhong — APPROVED** [review]

**Comment by Qiaolin-Yu** [comment]
failed test in https://github.com/sgl-project/sglang/actions/runs/26214938490/job/77457542284 has passed successfully after merging main
